clients/web/src/components/tauri/configuration-value.tsx

```diff
@@ -1,57 +1,43 @@
-import { Show, For, Switch, Match } from "solid-js";
+import { Show, For, createMemo } from "solid-js";
 import { ConfigurationTooltip } from "./configuration-tooltip";
 import { Flags } from "./configuration-value/flags";
+import { TauriConfig } from "~/lib/tauri/config/tauri-conf";
 
-type ConfigurationValue =
-  | ConfigurationRecord
-  | string
-  | []
-  | boolean
-  | null
-  | undefined
-  | unknown;
-
-/* eslint-disable-next-line @typescript-eslint/no-empty-interface */
-interface ConfigurationRecord extends Record<string, ConfigurationValue> {}
+type ConfigurationValue = TauriConfig[keyof TauriConfig];
 
 interface ConfigurationValueProps {
   parentKey: string;
   key: string;
   value: ConfigurationValue;
 }
 
-interface TextConfigurationValueProps extends ConfigurationValueProps {
+type TextConfigurationValueProps = Omit<ConfigurationValueProps, "value"> & {
   value: string | boolean;
-}
+};
 
 interface ArrayConfigurationValueProps extends ConfigurationValueProps {
-  value: [];
+  value: ConfigurationValue[];
 }
 
 interface ObjectConfigurationValueProps extends ConfigurationValueProps {
-  value: ConfigurationRecord;
+  value: Omit<ConfigurationValue, "undefined">;
```

Comment:
You should be able to use `Required<>` here (it's the same, just a bit more semantic ;) )

Reply:
That makes a lot of sense.